### PR TITLE
Added/updated cli test stubs for pxe-less discovery feature

### PR DESCRIPTION
**Pull request by sghai**

### CODE REVIEW

**Comment by elyezer** [comment]
ACK

File: tests/foreman/cli/test_discovery.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 
 class DiscoveryTestCase(CLITestCase):
     """Implements Foreman discovery CLI tests."""
 
+    @run_only_on('sat')
     @stubbed()
-    def test_positive_discovered_host_facts(self):
-        """@Test: Check all facts of discovered hosts are correctly displayed
+    @tier3
+    def test_custom_facts_pxeless_discovery(self):
```
Comment (kbidarkar) [inline tests/foreman/cli/test_discovery.py]: As per issue #3083 , we need to specify whether this is positive or negative testcase.
Hence the second word needs to be `test_positive` or `test_negative`.

File: tests/foreman/cli/test_discovery.py [hunk 2]
```diff
@@ -19,62 +22,132 @@ def test_positive_discovered_host_facts(self):
...
-        @Assert: All facts should be displayed correctly
+        @Assert: All defined custom facts should be displayed correctly
 
         @Status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_custom_facts_pxe_discovery(self):
```
Comment (kbidarkar) [inline tests/foreman/cli/test_discovery.py]: test_positive ??

File: tests/foreman/cli/test_discovery.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/cli/test_discovery.py]: :+1:

File: tests/foreman/cli/test_discovery.py [hunk 3]
```diff
@@ -19,62 +22,132 @@ def test_positive_discovered_host_facts(self):
...
+    @tier3
+    def test_positive_provision_pxe_host(self):
+        """Provision the pxe based discovered host from cli
+
+        @Feature: Foreman Discovery - PXEBased
 
+        @Setup: Host should already be discovered
+
+        @Assert: Host should be provisioned successfully and entry from
+        discovered host list should be auto removed
```
Comment (sthirugn) [inline tests/foreman/cli/test_discovery.py]: s/auto/automatically please

File: tests/foreman/cli/test_discovery.py [hunk 4]
```diff
@@ -19,62 +22,132 @@ def test_positive_discovered_host_facts(self):
...
+
+        @Status: Manual
         """
 
+    @run_only_on('sat')
     @stubbed()
-    def test_positive_refresh_discovered_host_facts(self):
-        """@Test: Refresh the facts of discovered hosts
+    @tier3
+    def test_positive_refresh_facts(self):
```
Comment (sthirugn) [inline tests/foreman/cli/test_discovery.py]: test_positive_refresh_facts_pxe_host?

File: tests/foreman/cli/test_discovery.py [hunk 5]
```diff
@@ -83,24 +156,67 @@ def test_positive_reboot_discovered_host(self):
...
 
         @Assert: Host should be successfully rebooted and provisioned
 
         @Status: Manual
+        """
 
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_list_discvered_host(self):
```
Comment (sthirugn) [inline tests/foreman/cli/test_discovery.py]: type - `discovered`

**Comment by sthirugn** [comment]
Pending comments and rebase

File: tests/foreman/cli/test_discovery.py (hunk 2 above)
Comment (sghai) [inline tests/foreman/cli/test_discovery.py]: sure

File: tests/foreman/cli/test_discovery.py (hunk 4 above)
Comment (sghai) [inline tests/foreman/cli/test_discovery.py]: :+1:

File: tests/foreman/cli/test_discovery.py (hunk 5 above)
Comment (sghai) [inline tests/foreman/cli/test_discovery.py]: :+1:

**Comment by sghai** [comment]
Rebased. Thank you for all comments.

**Comment by sghai** [comment]
Renamed file `test_discovery.py` to `test_discoveredhost.py`

**Comment by sthirugn** [comment]
ACK

**Pull request merged**